### PR TITLE
fix `masternode current` rpc

### PR DESCRIPTION
**Pull request by UdjinM6**
GetNextMasternodeInQueueForPayment should not filter scheduled masternodes for it

### CODE REVIEW

**Comment by codablock** [comment]
Is this a bugfix or a performance fix?

File: src/rpc/masternode.cpp [hunk 1]
```diff
@@ -189,27 +189,37 @@ UniValue masternode(const UniValue& params, bool fHelp)
...
-        if(!mnodeman.GetNextMasternodeInQueueForPayment(nHeight, true, nCount, mnInfo))
-            return "unknown";
+        if (strCommand == "winner" || !mnpayments.GetBlockPayee(nHeight, payee)) {
+            int nCount;
+            mnodeman.UpdateLastPaid(pindex);
+            if (!mnodeman.GetNextMasternodeInQueueForPayment(nHeight, true, nCount, mnInfo))
+                return "unknown";
+            txdest = mnInfo.pubKeyCollateralAddress.GetID();
+        } else {
+            if (!mnodeman.GetMasternodeInfo(payee, mnInfo))
```
Comment (codablock) [inline src/rpc/masternode.cpp]: It's not clear here that this depends on the result of GetBlockPayee. Any chance you restructure the if/else somehow that the 3 different cases we may have here are more clear? IMHO, in such cases minor code duplication is actually better then having hard to read if/else branches.

**Comment by UdjinM6** [comment]
It's a bug fix but I was fixing in the wrong place, thanks for pushing me to think harder :)
Pushed new version. It's also rebased on top of current 12.2.

**Comment by gladcow** [comment]
utACK

**Comment by the-schnibble** [comment]
utACK

**Comment by willwray** [comment]
utACK

**Pull request merged**